rotkehlchen/exchanges/ftx.py

```diff
@@ -0,0 +1,528 @@
+        """
+        request_verb = "GET"
+        backoff = self.initial_backoff
+
+        # Use a while loop to retry request if rate limit is reached
+        while True:
+            request_url = '/api/' + endpoint
+            options = {}
+            if limit is not None:
+                options['limit'] = limit
+            if start_time is not None:
+                options['start_time'] = start_time
+            if end_time is not None:
+                options['end_time'] = end_time
+
+            if len(options) != 0:
+                request_url += '?' + urlencode(options)
+
+            timestamp = int(time.time() * 1000)
+            signature_payload = f'{timestamp}{request_verb}{request_url}'.encode()
+
+            signature = hmac.new(self.secret, signature_payload, 'sha256').hexdigest()
+            log.debug('FTX API query', request_url=request_url)
+
+            self.session.headers.update({
+                'FTX-KEY': self.api_key,
+                'FTX-SIGN': signature,
+                'FTX-TS': str(timestamp),
+            })
+
+            full_url = self.base_uri + request_url
+            try:
+                response = self.session.get(full_url)
+            except requests.exceptions.RequestException as e:
+                raise RemoteError(f'FTX API request {full_url} failed due to {str(e)}') from e
+
+            if response.status_code == HTTPStatus.TOO_MANY_REQUESTS:
+                if backoff < self.backoff_limit:
+                    log.debug(
+                        f'FTX rate limit exceeded on endpoint {request_url}. Backing off',
+                        seconds=backoff,
+                    )
+                    gevent.sleep(backoff)
+                    backoff = backoff * 2
+                    continue
+            # We got a result here
+            break
+
+        if response.status_code == 401:
+            raise FTXLoginError(
+                f'Authentication to FTX failed for {request_url}. Verify the time of the machine'
+                f'and that the keys for the API are correct',
+            )
+
+        if response.status_code != 200:
+            raise RemoteError(
+                f'FTX query {full_url} responded with error status code: '
+                f'{response.status_code} and text: {response.text}',
+            )
+
+        try:
+            json_ret = jsonloads_dict(response.text)
+        except JSONDecodeError as e:
+            raise RemoteError(f'FTX returned invalid JSON response: {response.text}') from e
+
+        if 'result' not in json_ret:
+            raise RemoteError(f'FTX json response does not contain data: {response.text}')
+
+        return json_ret['result']
+
+    @overload
+    def _api_query(  # pylint: disable=no-self-use
+            self,
+            endpoint: Literal['wallet/all_balances'],
+            start_time: Timestamp = None,
+            end_time: Timestamp = None,
+            limit: int = 100,
```

Comment:
This `100` seems to be a default constant you use a lot in the module. Make it a constant on the top of the file.